packages/lab/src/carousel/CarouselReducer.ts

```diff
@@ -0,0 +1,94 @@
+import type { Dispatch } from "react";
+
+import type { CarouselSlideId, CarouselSlideMeta } from "./CarouselSlide";
+
+export interface CarouselReducerState {
+  slides: Map<CarouselSlideId, CarouselSlideMeta>;
+  activeSlideIndex: number;
+  visibleSlides: number;
+  focusedSlideIndex: number;
+  carouselId?: string;
+}
+export type CarouselReducerAction =
+  | { type: "register"; payload: [CarouselSlideId, CarouselSlideMeta] }
+  | { type: "unregister"; payload: CarouselSlideId }
+  | { type: "updateSlideCount"; payload: number }
+  | { type: "move"; payload: CarouselSlideId }
+  | { type: "moveToIndex"; payload: number }
+  | { type: "scroll"; payload: CarouselSlideId };
+
+export type CarouselReducerDispatch = Dispatch<CarouselReducerAction>;
+
+export function carouselReducer(
+  state: CarouselReducerState,
+  action: CarouselReducerAction,
+) {
+  switch (action.type) {
+    case "register": {
+      const { slides } = state;
+      const [id, { element, slideDescription }] = action.payload;
+      slides.set(id, { element, slideDescription });
+      return {
+        ...state,
+        slides: new Map(slides),
+      };
+    }
+    case "unregister": {
+      const { slides } = state;
+      const id = action.payload;
+      if (!slides.has(id)) return state;
+      slides.delete(id);
+      return {
+        ...state,
+        slides: new Map(slides),
+      };
+    }
+    // moves the first visible item
+    case "move": {
+      const { slides } = state;
+      const id = action.payload;
+      if (!slides.has(id)) return state;
+      const slideIds = [...slides.keys()];
+      const index = slideIds.indexOf(id || slideIds[0]);
+      return {
+        ...state,
+        activeSlideIndex: index,
+      };
+    }
+    case "moveToIndex": {
+      const { slides } = state;
+
+      const index = action.payload;
+
+      if (index === -1 || index > slides.size) return state;
```

Comment:
NIT do we enforce/prefer braces on single line conditions to create better readability ?